### ISSUE TITLE
JavadocTypeCheck: false positive with JavaDoc @param tags for Records (Java 14 + Java 15)

### ISSUE THREAD

**Issue by vovkss** [checkstyle/checkstyle#8808]
Check documentation: https://checkstyle.org/config_javadoc.html#JavadocType

```
/var/tmp $ javac MyRecord.java --enable-preview --release=15
Note: MyRecord.java uses preview language features.
Note: Recompile with -Xlint:preview for details.


/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocType">
      <property name="scope" value="protected"/>
      <property name="allowMissingParamTags" value="false"/>
      <property name="allowUnknownTags" value="false"/>
    </module>
  </module>
</module>


/var/tmp $ cat MyRecord.java
/**
 * My new record.
 *
 * @param value Sponge Bob rules the world!
 */
public record MyRecord(
    String value
) {
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.36-all.jar -c config.xml MyRecord.java
Starting audit...
[ERROR] C:\Users\vovks\Desktop\xx\MyRecord.java:4:4: Unused @param tag for '<value>'. [JavadocType]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

**The expectation is that no errors are reported** — `@param` is the way to document record fields. See https://stackoverflow.com/a/61451046/2170192
https://bugs.openjdk.java.net/browse/JDK-8225055

**Comment by romani** [comment]
@nmancus1 , can you show here how javadoc tool handle records(code and html screenshot) when defined in ctor and just fields of records.

**Comment by nmancus1** [comment]
```
/**
 * This is an example of a javadoc on a record.
 *
 * @param myString this is an example of a param tag used for a record component.
 * @param myInteger this is another example of a param tag used for a record component.
 * @param <X> my type parameter
 * @param <Y> my other type parameter
 */
public record JavadocRecord<X, Y>(String myString, Integer myInteger) {

    /**
     * Javadoc comment on constant
     */
    public static final int MY_INTEGER = 66;

    /**
     * Javadoc comment on field
     */
    private static int myInt = 42;

    /**
     *
     * @param myString my String
     * @param myInteger my Integer
     */
    public JavadocRecord {

    }

    /**
     * Javadoc comment on canonical ctor
     * @param myString the string to use
     */
    public JavadocRecord(String myString) {
        this(myString, MY_INTEGER);
    }

    /**
     * Javadoc comment on method
     *
     * @return the value of myInt
     */
    public static int getMyInt() {
        return myInt;
    }

    /**
     * Javadoc comment on method
     *
     * @param myInt the value to set
     */
    public static void setMyInt(int myInt) {
        JavadocRecord.myInt = myInt;
    }
}

```
![image](https://user-images.githubusercontent.com/31252532/93715483-84529880-fb37-11ea-9392-1b37fd622f6a.png)

**Comment by romani** [comment]
fix is merged.
